nginx/proxy.conf

```diff
@@ -1,3 +1,5 @@
+resolver 127.0.0.1;
```

Comment:
It wasn't obvious to me too when I saw nginx complaining that it couldn't resolve the host ;-)

Btw, I forgot to mention that I tested this setup with Nginx 1.8.0. And I'm only using the maven repo, hopefully the other ones should work the same.